packages/mermaid/test/clean.spec.ts

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+import { MermaidManager } from '@jupyterlab/mermaid';
+
+import {
+  makeElementVariants,
+  SVG_MIME,
+  SVG_XHTML_FOOTER,
+  SVG_XHTML_HEADER,
+  VOID_ELEMENTS
+} from './utils';
+
+const parser = new DOMParser();
+
+describe('@jupyterlab/mermaid', () => {
+  for (const element of VOID_ELEMENTS) {
+    it(`should clean void element ${element}`, () => {
+      const variants = makeElementVariants(element);
+      const raw = [...SVG_XHTML_HEADER, ...variants, ...SVG_XHTML_FOOTER].join(
+        '\n'
+      );
+
+      const clean = MermaidManager.cleanMermaidSvg(raw);
+      const parsed = parser.parseFromString(clean, SVG_MIME);
```

Comment:
This would now error if the cleaned string is not valid XML... it's harder to verify that the browser would go into the "fallback" rendering mode (the most likely failure case).

Because of that, it's probably _not_ worth roundtripping inside the implementation, which is already pretty heavy.